lib/worker.py

```diff
@@ -326,7 +328,19 @@ def run_loop(self, task_queue, result_queue):
                 break
 
             result_queue.put(self.current_task(task_id))
-            short_status = self.run_task(task_id)
+            short_status = ''
+            # 'fragile_retries' value comes from the suite configuration
+            retries_left = fragile_retries
+            while short_status != 'pass' and retries_left >= 0:
```

Comment:
Must be used only for "fragile" tests. IMHO, retries_left > 0 looks more correct.
May be something like:
run_test()
if no pass and fragile then retry...

Reply:
Split into 2 parts for fragile and not fragile.